malibacollab/aws-kvs-webrtc-stream-dockerised
# kvs webrtc dockerized

### Description

This pull request adds a Dockerized version of the AWS KVS WebRTC C SDK integrated with GStreamer. It allows users to stream video and audio to AWS KVS WebRTC using customizable GStreamer pipelines, supporting various input sources like USB cameras and RTSP streams (IP cameras). 

**Key Features:**
- Customizable GStreamer pipelines for different video/audio sources.
- Dockerized for easy deployment and consistent environment setup.
- Integrated with AWS KVS WebRTC for real-time streaming.
- Audio/video capture and processing through GStreamer.

### Changes Included
- Docker setup (`Dockerfile`, `docker-compose.yml`) to run the AWS KVS WebRTC C SDK.
- Configuration for GStreamer pipelines via the `PIPELINE` environment variable.
- Added documentation in the README for setup, configuration, and usage examples.

### How to Test
1. Build the Docker image with `docker-compose build`.
2. Set up AWS credentials and the necessary certificates.
3. Modify the GStreamer `PIPELINE` in `docker-compose.yml` to match the video source (USB, RTSP, etc.).
4. Start the container with `docker-compose up` and verify the stream in AWS KVS WebRTC.

### Related Issues
- N/A (or list any relevant issue IDs).
